### DISCUSSION TITLE
Why is permission to open downloaded files necessary?

### DISCUSSION THREAD

**Discussion by SquashSquishSquaw** [aclap-dev/video-downloadhelper#716] [Help and Q&A] [answered]
### Your question

Hello,

Firefox just now (May 27, 2024) informed me that there is a new version of video downloadhelper ready for download, but it needs permission to open the downloaded files. Why does it need that?

Many thanks!

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"C:\\Program Files\\net.downloadhelper.coapp\\bin\\net.downloadhelper.coapp-win-64.exe","version":"1.6.3","new_version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0
{
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 3
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I have never had that problem. Please post some screenshots of what you are seeing.

I have never dealt with a squishy squaw before.  I hope I'm not tempting fate.  I have, however, played squash.  Even eaten a few.  That was pretty safe.

**Comment by paulrouget** [comment, marked as answer]
> Firefox just now (May 27, 2024) informed me that there is a new version of video downloadhelper ready for download, but it needs permission to open the downloaded files. Why does it need that?

When you press the "Play" button after a download, we use that new API to play that file with the native video player.

Here is the documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/open

**Reply by GrampaWildWilly** [reply]
Oh really?  So it no longer just queues the video up in the VLC playlist?  I'll have to experiment with that.

**Reply by GrampaWildWilly** [reply]
Nope.  All the green play button does is queue the file up for VLC.  Maybe there's some configuation parameter for getting it to do what you describe, Paul.

**Reply by GrampaWildWilly** [reply]
Oh.  Wait.  I do have a file association for MP4 to VLC.  I'd have to break that to get the behavior you're describing.  Never mind.

**Reply by paulrouget** [reply]
@GrampaWildWilly that's for when you don't have a coapp.

**Reply by SquashSquishSquaw** [reply]
Thanks for the explanation, that's great!
